host/xtest/regression_1000.c

```diff
@@ -2687,3 +2694,343 @@ static void xtest_tee_test_1036(ADBG_Case_t *c)
 }
 ADBG_CASE_DEFINE(regression, 1036, xtest_tee_test_1036,
 		 "Test PAuth (Pointer Authentication)");
+
+#define MAX_KEY_SIZE 4096
+
+#ifdef OPENSSL_FOUND
+static RSA *att_key;
+size_t att_key_size;  /* Actual key size (modulus size) in bytes */
+
+/*
+ * buf = [ TA hash (32 bytes SHA256) | Signature (att_key_size bytes) ]
+ * Signature = RSA_SSA_PKCS1_PSS_MGF1(SHA256([nonce | TA hash])
+ */
+static void check_signature(ADBG_Case_t *c, uint8_t *nonce, size_t nonce_size,
+			    uint8_t *buf)
+{
+	unsigned char digest[SHA256_DIGEST_LENGTH] = { };
+	unsigned char *ta_hash = buf;
+	unsigned char *sig = buf + SHA256_DIGEST_LENGTH;
+	int salt_len = 32; /* Hard-coded in the PTA */
+	uint8_t decr[MAX_KEY_SIZE / 8] = { };
+	SHA256_CTX ctx = { };
+
+	ADBG_EXPECT_NOT_NULL(c, att_key);
+	if (!att_key)
+		return;
+
+	SHA256_Init(&ctx);
+	SHA256_Update(&ctx, nonce, nonce_size);
+	SHA256_Update(&ctx, ta_hash, SHA256_DIGEST_LENGTH);
+	SHA256_Final(digest, &ctx);
+
+	ADBG_EXPECT_COMPARE_SIGNED(c, RSA_public_decrypt(att_key_size, sig,
+						decr, att_key,
+						RSA_NO_PADDING),
+				   >, 0);
+	ADBG_EXPECT_COMPARE_SIGNED(c, RSA_verify_PKCS1_PSS_mgf1(att_key, digest,
+						EVP_sha256(), EVP_sha256(),
+						decr, salt_len),
+				   >, 0);
+}
+
+static void free_att_key(void)
+{
+	RSA_free(att_key);
+	att_key = NULL;
+}
+
+static void set_att_key(ADBG_Case_t *c, uint8_t *e, size_t e_sz, uint8_t *n,
+			size_t n_sz)
+{
+	int st = 0;
+	BIGNUM *bn_e = NULL;
+	BIGNUM *bn_n = NULL;
+
+	att_key_size = n_sz;
+	att_key = RSA_new();
+	ADBG_EXPECT_NOT_NULL(c, att_key);
+
+	ADBG_EXPECT_NOT_NULL(c, bn_e = BN_bin2bn(e, e_sz, NULL));
```

Comment:
Please do the assignment first on one line and then this check on the next.
Same comment for a few other places.